pkg/server/authentication.go

```diff
@@ -227,17 +227,17 @@ WHERE id = $1`
 	}
 
 	if row.Len() != 4 ||
-		row[0].ResolvedType() != types.Bytes ||
-		row[1].ResolvedType() != types.String ||
-		row[2].ResolvedType() != types.Timestamp {
+		!types.EqualTypes(row[0].ResolvedType(), types.Bytes) ||
```

Comment:
Almost every case you've replaced will work fine as-is. The `==` operator only fails if the *both* sides are `TTuple`. So if we have `types.String` on one side, we're fine, even if the other side is `TTuple`. Here's some code that illustrates how `==` works:
```
type tUnk struct {
}

type tTuple struct {
	Types []Type
}

type Type interface {
}

func main() {
	var i, i2 Type
	i = tUnk{}
	i2 = tTuple{Types: []Type{i}}

	// true
	fmt.Println(i == i)
	
	// false, even though one side is tTuple
	fmt.Println(i == i2)
	
	// panic: runtime error: comparing uncomparable type main.tTuple
	fmt.Println(i2 != i2)
}
```

Reply:
Btw, I can't recall where I saw this technique, but we could structure `TTuple` so that `==` works. The basic idea is to use a linked list instead of a slice. See https://play.golang.org/p/xpPFNpGlxtR. That example doesn't handle arbitrary nesting, but I don't think that would be difficult to add.

Reply:
I like that idea a lot. `TTuple` is the only (unusual) case where we have an issue, so better to push complexity into that type rather than forcing every user to deal with it (or not, and introduce bugs as a result).

Reply:
Here's an extension to arbitrary nesting: https://play.golang.org/p/Sk_zkW7Drid